### PR TITLE
Add Spanish translation of custom_models.mdx

### PR DESCRIPTION
**Pull request by donelianc**
# What does this PR do?

Add the Spanish translation for `custom_models.mdx` as part of the #15947 issue. 
Changes include the Spanish version of the original document and the updated `_toctree.yml` file.

Tried to follow the [guideline](https://github.com/huggingface/transformers/blob/26a6a426087582c48593f8be980603951a7bcddd/CONTRIBUTING.md#start-contributing-pull-requests) to generate the Markdown files and check them before submitting this PR but the command `doc-builder build transformers docs/source/ --build_dir ~/tmp/test-build` fails since the `_toctree.yml` file is no longer in `./docs/source/` (but in `./docs/source/en/`).

First contribution to the 🤗 Transformers project!

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests) Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Task assignment [here](https://github.com/huggingface/transformers/issues/15947#issuecomment-1161854258).
- [x] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] ~~Did you write any new necessary tests?~~

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Tagging @omarespejel, @osanseviero, or @sgugger to review or assign reviewers :)

### CODE REVIEW

**Comment by osanseviero** [comment]
@omarespejel could you take a look at this PR Please?

File: docs/source/es/custom_models.mdx [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+La biblioteca 🤗 Transformers está diseñada para ser fácilmente ampliable. Cada modelo está completamente codificado 
+sin abstracción en una subcarpeta determinada del repositorio, por lo que puedes copiar fácilmente un archivo del modelo 
+y ajustarlo según tus necesidades.
+
+Si estás escribiendo un modelo completamente nuevo, podría ser más fácil comenzar desde cero. En este tutorial, te mostraremos 
+cómo escribir un modelo personalizado y su configuración para que pueda usarse dentro de Transformers, y cómo puedes compartirlo 
+con la comunidad (con el código en el que se basa) para que cualquiera pueda usarlo, incluso si no está presente en la biblioteca 
+🤗 Transformers.
+
+Ilustraremos todo esto en un modelo ResNet, envolviendo la clase ResNet del [biblioteca timm](https://github.com/rwightman/pytorch-image-models/tree/master/timm) en un [`PreTrainedModel`].
```
Comment (omarespejel) [inline docs/source/es/custom_models.mdx:24]:
```suggestion
Ilustraremos todo esto con un modelo ResNet, envolviendo la clase ResNet de la [biblioteca timm](https://github.com/rwightman/pytorch-image-models/tree/master/timm) en un [`PreTrainedModel`].
```

File: docs/source/es/custom_models.mdx [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+## Escribir una configuración personalizada
+
+Antes de adentrarnos en el modelo, primero escribamos su configuración. La configuración de un modelo es un objeto que
+contendrá toda la información necesaria para construir el modelo. Como veremos en la siguiente sección, el modelo solo puede
+tomar un `config` para ser inicializado, por lo que realmente necesitamos que ese objeto esté lo más completo posible.
+
+En nuestro ejemplo, tomaremos un par de argumentos de la clase ResNet que tal vez queramos modificar. Las diferentes 
+configuraciones nos darán los diferentes tipos de ResNet que son posibles. Luego simplemente almacenamos esos argumentos,
```
Comment (omarespejel) [inline docs/source/es/custom_models.mdx:33]:
```suggestion
configuraciones nos darán los diferentes tipos de ResNet que son posibles. Luego simplemente almacenamos esos argumentos 
```

File: docs/source/es/custom_models.mdx [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+```
+
+Esto guardará un archivo llamado `config.json` dentro de la carpeta `custom-resnet`. Luego puedes volver a cargar tu configuración 
+con el método `from_pretrained`:
+
+```py
+resnet50d_config = ResnetConfig.from_pretrained("custom-resnet")
+```
+
+También puedes usar cualquier otro método de la clase [`PretrainedConfig`], como [`~PretrainedConfig.push_to_hub`] para cargar
```
Comment (omarespejel) [inline docs/source/es/custom_models.mdx:101]:
```suggestion
También puedes usar cualquier otro método de la clase [`PretrainedConfig`], como [`~PretrainedConfig.push_to_hub`], para cargar 
```

**Review by omarespejel — CHANGES_REQUESTED** [review]

File: docs/source/es/custom_models.mdx [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+from transformers import AutoConfig, AutoModel, AutoModelForImageClassification
+
+AutoConfig.register("resnet", ResnetConfig)
+AutoModel.register(ResnetConfig, ResnetModel)
+AutoModelForImageClassification.register(ResnetConfig, ResnetModelForImageClassification)
+```
+
+Ten en cuenta que el primer argumento utilizado al registrar tu configuración personalizada en [`AutoConfig`] debe coincidir
+con el `model_type` de tu configuración personalizada, y el primer argumento utilizado al registrar tus modelos personalizados
+en cualquier clase del modelo automático debe coincidir con `config_class ` de esos modelos.
```
Comment (omarespejel) [inline docs/source/es/custom_models.mdx:351]:
```suggestion
en cualquier clase del modelo automático debe coincidir con el `config_class` de esos modelos.
```

File: docs/source/es/custom_models.mdx [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+obtener más información), pero aún debes revisar el código del modelo y el autor para evitar la ejecución de código malicioso 
+en tu computadora. Configura `trust_remote_code=True` para usar un modelo con código personalizado:
+
+```py
+from transformers import AutoModelForImageClassification
+
+model = AutoModelForImageClassification.from_pretrained("sgugger/custom-resnet50d", trust_remote_code=True)
+```
+
+También se recomienda encarecidamente pasar un _hash_ de confirmación como una "revisión" para asegurarte que el autor
```
Comment (omarespejel) [inline docs/source/es/custom_models.mdx:318]:
```suggestion
También se recomienda encarecidamente pasar un _hash_ de confirmación como una "revisión" para asegurarte de que el autor 
```

**Review by omarespejel — CHANGES_REQUESTED** [review]

**Comment by omarespejel** [comment]
Hola @donelianc! Thank you very much for your translation. Sorry for the delay in replying; it won't happen again.

I made some small comments to be applied.

Thank you!

**Comment by donelianc** [comment]
@omarespejel I committed the suggested changes :) If you don't mind, can you assign me the translation of [run_scripts.mdx](https://github.com/huggingface/transformers/blob/b9a768b3ffa80c4c19d024f9f42d5917e7d8109e/docs/source/en/run_scripts.mdx)? I'll be happy to help with one more doc.

**Review by sgugger — APPROVED** [review]
Thanks a lot for your contribution!

**Pull request merged**